File: extract_js.py

import fs

with open('index.html', 'r', encoding='utf-8') as f:
    lines = f.readlines()

start_index = -1
end_index = -1

# Find the script block. We know it starts around line 195 and ends around line 4080.
# We look for the <script> tag that starts the main js logic.
# Line 195 was <script>
# Line 4080 was </script>

for i, line in enumerate(lines):
    if '<script>' in line and i > 150: # Avoid earlier scripts
        start_index = i
        break

for i in range(len(lines) - 1, -1, -1):
    if '</script>' in lines[i]:
        end_index = i
        break

if start_index != -1 and end_index != -1:
    js_content = lines[start_index + 1 : end_index]
    
    # Save js content to js/app.js
    with open('js/app.js', 'w', encoding='utf-8') as f_js:
        f_js.writelines(js_content)
    
    # Replace the script block in index.html
    new_lines = lines[:start_index] + ['    <script type="module" src="js/app.js"></script>\n'] + lines[end_index + 1:]
    
    with open('index.html', 'w', encoding='utf-8') as f_html:
        f_html.writelines(new_lines)
    
    print(f"Extracted JS from line {start_index + 1} to {end_index + 1} into js/app.js")
else:
    print("Could not find script block")

File: extract_js.js

const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'index.html');
const jsPath = path.join(__dirname, 'js/app.js');

const htmlContent = fs.readFileSync(filePath, 'utf-8');

// Find the script tag that starts at line 195 (around)
// We look for <script> after line 180...
const lines = htmlContent.split('\n');

let startIndex = -1;
let endIndex = -1;

for (let i = 0; i < lines.length; i++) {
    if (lines[i].trim() === '<script>' && i > 180) {
        startIndex = i;
        break;
    }
}

for (let i = lines.length - 1; i >= 0; i--) {
    if (lines[i].trim() === '</script>') {
        endIndex = i;
        break;
    }
}

if (startIndex !== -1 && endIndex !== -1) {
    // Extract JS content
    const jsContent = lines.slice(startIndex + 1, endIndex).join('\n');

    // Write JS file
    fs.mkdirSync(path.dirname(jsPath), { recursive: true });
    fs.writeFileSync(jsPath, jsContent);

    // Replace script tag in HTML
    const newHtmlLines = [
        ...lines.slice(0, startIndex),
        '    <script type="module" src="js/app.js"></script>',
        ...lines.slice(endIndex + 1)
    ];

    fs.writeFileSync(filePath, newHtmlLines.join('\n'));

    console.log(`Extracted JS from lines ${startIndex + 1} to ${endIndex + 1} into js/app.js`);
} else {
    console.error('Could not find script block');
}
